src/app_service/service/message/service.nim

```diff
@@ -170,32 +229,44 @@ QtObject:
     # if (not chats[0].active):
     #   return
 
-    for msg in messages:
-      if(msg.editedAt > 0):
-        let data = MessageEditedArgs(chatId: msg.localChatId, message: msg)
-        self.events.emit(SIGNAL_MESSAGE_EDITED, data)
-
     for i in 0 ..< chats.len:
+      let chatId = chats[i].id
+
       if(chats[i].chatType == ChatType.Unknown):
-        error "error: new message with an unknown chat type received", chatId=chats[i].id
+        error "error: new message with an unknown chat type received", chatId=chatId
         continue
 
       # Ignore 1-1 chats for which we are not contact
-      if (chats[i].chatType == ChatType.OneToOne and not self.contactService.getContactById(chats[i].id).isContact):
+      if(chats[i].chatType == ChatType.OneToOne and not self.contactService.getContactById(chatId).isContact):
         continue
 
+      let currentChatCursor = self.initOrGetMessageCursor(chatId)
+      # Ignore messages update if chat haven't loaded any messages and try to load them from database instead
+      if(currentChatCursor.isEmpty()):
+        currentChatCursor.makeObsolete()
+        self.asyncLoadMoreMessagesForChat(chatId)
+        continue
```

Comment:
What happens if one of the message is an edit or delete, would they be applied when scrolling up?

Reply:
In this case, when the chat cursor is empty, messages are not yet loaded, meaning edit or delete can't be applied, because there are no messages it can refer to. When scrolling up, the messages will be loaded from the database and is up to status-go to handle edits and deletes properly.

Reply:
That makes sense!